core/src/main/java/org/elasticsearch/node/NodeModule.java

```diff
@@ -58,5 +97,30 @@ protected void configure() {
         bind(Node.class).toInstance(node);
         bind(MonitorService.class).toInstance(monitorService);
         bind(NodeService.class).asEagerSingleton();
+        bind(ProcessorsRegistry.class).toInstance(processorsRegistry);
+    }
+
+    /**
+     * Returns the node
+     */
+    public Node getNode() {
+        return node;
+    }
+
+    /**
+     * Adds a processor factory under a specific type name.
+     */
+    public void registerProcessor(String type, Function<TemplateService, Processor.Factory<?>> processorFactoryProvider) {
+        processorsRegistry.registerProcessor(type, processorFactoryProvider);
+    }
+
+    public static boolean isNodeIngestEnabled(Settings settings) {
+        return settings.getAsBoolean("node.ingest", true);
+    }
+
+    public static boolean isNodeIngestEnabled(ImmutableOpenMap<String, String> nodeAttributes) {
+        String ingestEnabled = nodeAttributes.get("ingest");
+        //reproduces same logic used in settings.getAsBoolean used above
+        return Booleans.parseBoolean(ingestEnabled, true);
```

Comment:
oh sorry I was totally not aware of this new Setting stuff I was referring to Settings.getBoolean . Thanks for explaining